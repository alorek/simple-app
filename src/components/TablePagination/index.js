import React, { Component } from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import times from 'lodash/times';
import { selectPage, nextPage, previousPage } from '../../actions';

class TablePagination extends Component {
    render() {
        const currentPage = this.props.currentPage;
        const self = this;
        const pages = times((this.props.data.length / 5), function(i) {
            const page = i + 1;
            const isActive = page === currentPage;

            return (
                <li
                key={i}
                onClick={ () => { self.props.selectPage(page); } }
                className= { isActive ? 'active' : ''} ><a>
                    { page }
                </a></li>
            );
        });

        return (
            <ul className="pagination pull-right">
                <li
                    onClick={ () => { this.props.previousPage(); } }>
                    <a>&laquo;</a>
                </li>
                { pages }
                <li
                    onClick={ () => { this.props.nextPage(); } }>
                    <a>&raquo;</a>
                </li>
            </ul>
        );
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        selectPage: selectPage,
        nextPage: nextPage,
        previousPage: previousPage
    }, dispatch);
}

function mapStateToProps(state){
    return {
        currentPage: state.table.currentPage,
        data: state.table.data
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(TablePagination);
