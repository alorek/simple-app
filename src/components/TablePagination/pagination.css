

.table-pagination {
	border: 1px solid black;
}

.table-pagination span {
	    border-radius: 86px;
    background: #999;
    width: 30px;
    height: 30px;
    display: inline-block;
    line-height: 30px;
    color: #fff;
}

.table-pagination span.active {
    background-color: blue;
    font-weight: bold;
}
