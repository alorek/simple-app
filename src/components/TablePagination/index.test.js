import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import TablePagination from './index';
import reducer from '../../reducers';

const store = createStore(reducer);

test('TablePagination renders correctly', () => {

    const tree = renderer.create(
        <Provider store={ store }>
            <TablePagination />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
