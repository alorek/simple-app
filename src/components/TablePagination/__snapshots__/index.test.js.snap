// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePagination renders correctly 1`] = `
<ul
  className="pagination"
>
  <li
    onClick={[Function]}
  >
    <a>
      «
    </a>
  </li>
  <li
    className="active"
    onClick={[Function]}
  >
    <a>
      1
    </a>
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    <a>
      2
    </a>
  </li>
  <li
    onClick={[Function]}
  >
    <a>
      »
    </a>
  </li>
</ul>
`;
